cmd/snapctl/main_test.go

```diff
@@ -47,6 +47,8 @@ var _ = Suite(&snapctlSuite{})
 
 func (s *snapctlSuite) SetUpTest(c *C) {
 	os.Setenv("SNAP_COOKIE", "snap-context-test")
+	// don't use SNAP_CONTEXT, in case other tests accidentally leak this
+	os.Unsetenv("SNAP_CONTEXT")
```

Comment:
This looks fine, it shouldn't be a problem if we unset it in the test binary.

Still, I don't see it being set anywhere in the package. AFAIK, when running go test ./..., each package would get built and run separately, so even if a test in another package sets SNAP_CONTEXT, this should not leak into the test in another package. Is it possible that the tests were executed by Go installed from a snap?

Reply:
I agree, it's weird, it would be nice to know where did it come from.

That being said, the change looks ok.

Reply:
> Is it possible that the tests were executed by Go installed from a snap?

Yes this is exactly what I did, for some reason it doesn't happen when I just use `./run-checks --unit` instead

> Still, I don't see it being set anywhere in the package

FWIW it is still set by snap-confine here: 

https://github.com/snapcore/snapd/blob/9b212f0a9f5afb81ab28db632807c9402a69b780/cmd/snap-confine/snap-confine.c#L462-L463